### ISSUE TITLE
Include training time arg

### ISSUE THREAD

**Issue by Burhan-Q** [ultralytics/ultralytics#6919]
## Summary
Two possible routes to use. 
1. **Primary:** Estimate the average duration required per training epoch to update LR-scheduler to finalize training on epoch number that is closest to the `time` provided.

    - EX:
        `yolo train time=3600 # arg example` 
        - if first 5 epochs average ~5 minutes/epoch, update LR-scheduler to finish at epoch `12` instead of default `100`

1. **Secondary:** Use `time` argument to 'interrupt' training (same as `CTRL+C`) after `t` training duration. **The model at this stage will be incomplete and NOT ready for use.** The goal with this method is to allow for a type of orchestration, where the model can be trained for an extended amount of time in smaller chunks if compute resources or availability are constrained.

    - see #6750 for additional details and conversation

## Requirements
- minimal modifications to training code
- if possible, only add single argument (maybe only possible per method)
- where feasible, reuse `EarlyStopping` class

**Comment by Burhan-Q** [comment]
linked #7054
